go/libkb/chain_link_v2_test.go

```diff
@@ -110,3 +111,57 @@ func TestOuterLinkV2WithMetadataPointerContainerDecode(t *testing.T) {
 	err = MsgpackDecode(&o, bytes)
 	requireErrorHasSuffix(t, errCodecDecodeSelf, err)
 }
+
+func serdeOuterLink(t *testing.T, tc TestContext, highSkipSeqno *keybase1.Seqno, highSkipHash LinkID) OuterLinkV2 {
+	m := NewMetaContextForTest(tc)
+
+	AddEnvironmentFeatureForTest(tc, EnvironmentFeatureAllowHighSkips)
+
+	var highSkip *HighSkip
+	highSkip = nil
+	if highSkipSeqno != nil {
+		highSkipPre := NewHighSkip(*highSkipSeqno, highSkipHash)
+		highSkip = &highSkipPre
+	}
+	encoded, err := encodeOuterLink(m, LinkTypeLeave, keybase1.Seqno(20), nil, nil, false, keybase1.SeqType_PUBLIC, false, highSkip)
+	require.NoError(t, err)
+
+	o2 := OuterLinkV2{}
+	err = MsgpackDecode(&o2, encoded)
+	require.NoError(t, err)
+	return o2
+}
+
+func highSkipMatch(ol OuterLinkV2, highSkip *HighSkip) error {
+	return ol.AssertFields(ol.Version, ol.Seqno, ol.Prev, ol.Curr, ol.LinkType, ol.SeqType, ol.IgnoreIfUnsupported, highSkip)
+}
+
+func TestHighSkipBackwardsCompatibility(t *testing.T) {
+	tc := SetupTest(t, "test_high_skip_backwards_compatibility", 1)
+	defer tc.Cleanup()
+
+	o := serdeOuterLink(t, tc, nil, nil)
+	require.True(t, o.HighSkipSeqno == nil)
+	require.True(t, o.HighSkipHash == nil)
+	require.NoError(t, highSkipMatch(o, nil))
+
+	seqno := keybase1.Seqno(3)
+	o = serdeOuterLink(t, tc, &seqno, nil)
+	require.True(t, *o.HighSkipSeqno == 3)
+	require.True(t, o.HighSkipHash == nil)
```

Comment:
this can't actually happen though: a highskipseqno of 3 and a highskiphash of nil. i'd argue we should use realistic cases for our happypath tests even if the code being tested doesn't verify it.